Gems/Atom/Tools/MaterialEditor/Code/Source/MaterialEditorApplication.cpp

```diff
@@ -210,96 +139,24 @@ namespace MaterialEditor
 
-        QStringList failedAssets;
-
-        // Forced asset processor to synchronously process all critical assets
-        // Note: with AssetManager's current implementation, a compiled asset won't be added in asset registry until next system tick. 
-        // So the asset id won't be found right after CompileAssetSync call. 
-        for (const AZStd::string& assetFilters : assetFiltersArray)
-        {
-            AZ_TracePrintf("MaterialEditor", "Compiling critical asset matching: %s.\n", assetFilters.c_str());
-
-            // Wait for the asset be compiled
-            AzFramework::AssetSystem::AssetStatus status = AzFramework::AssetSystem::AssetStatus_Unknown;
-            AzFramework::AssetSystemRequestBus::BroadcastResult(
-                status, &AzFramework::AssetSystemRequestBus::Events::CompileAssetSync, assetFilters);
-            if (status != AzFramework::AssetSystem::AssetStatus_Compiled)
-            {
-                failedAssets.append(assetFilters.c_str());
-            }
-        }
-
-        if (!failedAssets.empty())
-        {
-            QMessageBox::critical(activeWindow(),
-                QString("Failed to compile critical assets"),
-                QString("Failed to compile the following critical assets:\n%1\n%2")
-                .arg(failedAssets.join(",\n"))
-                .arg("Make sure this is an Atom project."));
-            ExitMainLoop();
-        }
-    }
-
-    void MaterialEditorApplication::SaveSettings()
-    {
-        if (m_activatedLocalUserSettings)
-        {
-            AZ::SerializeContext* context = nullptr;
-            AZ::ComponentApplicationBus::BroadcastResult(context, &AZ::ComponentApplicationRequests::GetSerializeContext);
-            AZ_Assert(context, "No serialize context");
-
-            char resolvedPath[AZ_MAX_PATH_LEN] = "";
-            AZ::IO::FileIOBase::GetInstance()->ResolvePath("@user@/MaterialEditorUserSettings.xml", resolvedPath, AZ_ARRAY_SIZE(resolvedPath));
-            m_localUserSettings.Save(resolvedPath, context);
-        }
-    }
-
-    bool MaterialEditorApplication::OnOutput(const char* window, const char* message)
-    {
-        // Suppress spam from the Source Control system
-        if (0 == strncmp(window, AzToolsFramework::SCC_WINDOW, AZ_ARRAY_SIZE(AzToolsFramework::SCC_WINDOW)))
+        if (connectedToAssetProcessor)
         {
-            return true;
+            CompileCriticalAssets(assetFiltersArray);
         }
 
-        if (m_logFile)
-        {
-            m_logFile->AppendLog(AzFramework::LogFile::SEV_NORMAL, window, message);
-        }
-        else
-        {
-            m_startupLogSink.push_back({ window, message });
-        }
-        return false;
+        AzFramework::AssetSystemStatusBus::Handler::BusDisconnect();
     }
 
     void MaterialEditorApplication::ProcessCommandLine(const AZ::CommandLine& commandLine)
```

Comment:
Everything except for the open document command line option handling can be moved into the base class process command line function.  

This function is also not calling the base class version.